### DISCUSSION TITLE
A way for publisher to not get published message?

### DISCUSSION THREAD

**Discussion by gitcatrat** [uNetworking/uWebSockets.js#435] [Q&A] [answered]
Thanks for making this wonderful library, I cannot believe I almost went with `ws`. This makes me want to learn C so bad.

Hypothetical scenario:
* user logs into chat app
* navigates to `/group/123`
* subscibes to `/group/123` `uws` topic
* sends a mesage (ideally would display if user hits Send, client already has data!)
* message gets `publish'd` and sent to everybody in group, **including himself**

Is there any way to exclude himself from receiving the message? 

I'd rather not maintain my own topics and subscribers in Node land, that would probably be a lot worse than extra message operational and network overhead. It could add up a lot though if I used this pattern a lot.

**Comment by ghost** [comment, marked as answer]
I agree this is an important feature. It is planned for next major version. Right now you cannot exclude the sender.

**Reply by gitcatrat** [reply]
Awesome to hear! Do you have any approximate timeframe for next major version?

**Reply by Xetera** [reply]
Would be super useful to have this feature out. Is it still planned?

**Reply by hst-m** [reply]
https://github.com/uNetworking/uWebSockets/releases/tag/v19.0.0

**Reply by Xetera** [reply]
Ah I didn't realize the behavior was changed instead of adding a new method. Thanks for the reply

**Comment by getTobiasNielsen** [comment]
Would sending a nonce to the server, and then back again work for you?

If the message is equal to the current nonce on the client, then the message is discarded.
